core/storage/feedback/gae_models.py

```diff
@@ -304,7 +304,26 @@ class UnsentFeedbackEmailModel(base_models.BaseModel):
     # Each element in this list is a dict with keys 'exploration_id',
     # 'thread_id' and 'message_id'; this information is used to retrieve
     # corresponding FeedbackMessageModel instance.
-    feedback_messages = ndb.JsonProperty(repeated=True)
+    feedback_message_references = ndb.JsonProperty(repeated=True)
     # The number of failed attempts that have been made (so far) to
     # send an email to this user.
     retries = ndb.IntegerProperty(default=0, required=True, indexed=True)
+
+    @classmethod
+    def create_or_modify(cls, owner_id, feedback_message_reference):
+        """Creates a new instance of UnsentFeedbackEmailModel for sending
```

Comment:
Can you move this logic to feedback_services and have it as a private function within the relevant function, similar to what we do in feedback_jobs_continuous.py, for example? I think that's more consistent, and also avoids adding too much logic to the storage layer. I also think that maybe it's better to call the private function something like _add_feedback_message_reference(user_id, reference).